### ISSUE TITLE
ArgumentOutOfRangeException if you put IgnoreMap attribute on a destination type property that there is no equivalent property to it in the source type

### ISSUE THREAD

**Issue by mjalil** [AutoMapper/AutoMapper#2266]
This might be 2 issues, but since both of them are from a single line of code, I mention them together:
1- Silently ignores mapping in the reverse direction (although, this behavior has some advocates)
2- throws an exception if you put `IgnoreMap` attribute on a destination type property that there is no equivalent property to it in the source type
The cause: https://github.com/AutoMapper/AutoMapper/blame/49e81dff7801259be2779f3054161c3a8b6cedaf/src/AutoMapper/Internal/MappingExpression.cs#L616
### Source/destination types

```csharp
public class BlogPost
{
    public long Id {get; set;}
    public string Body {get; set;}
    public DateTime PublishDate {get; set;}
    public string Publisher {get; set;}
}
public class BlogPostDto
{
    public long Id {get; set;}
    public string Body {get; set;}
    public DateTime PublishDate {get; set;}
    [IgnoreMap]
    public string Publisher {get; set;}
    [IgnoreMap]
    public List<BlogPostDto> RelatedPosts {get; set;}
}
```

### Mapping configuration

```csharp
Mapper.Initialize(cfg => {
    cfg.CreateMap<BlogPost, BlogPostDto>();
});
```

### Expected behavior

Ignore map `BlogPost.Publisher -> BlogPostDto.Publisher` but not ignore `BlogPostDto.Publisher -> BlogPost.Publisher`. also, IgnoreMap on `RelatedPosts` should not throw an exception, silently ignore it.
I just want to permanently ignore mapping of RelatedPosts now and at the future.
It was working In version 2.2 of the library as I expected. I think it is ok in 4.2.1 too. but not after v5.0.0-beta-1
### Actual behavior

silently ignores mapping of Publisher in the reverse direction.
and throws an ArgumentOutOfRangeException: 'Cannot find a field or property named RelatedPosts'
Now, after upgrading from 2.2 to 6.1.1 I have to search the source code and delete `IgnoreMap`s or initialize the mapper like this:
```csharp
Mapper.Initialize(cfg => {
    cfg.CreateMap<BlogPost, BlogPostDto>()
    .ReverseMap()
    .ForMember(entity => entity.Publisher, opt => opt.MapFrom(dto => dto.Publisher));
});
```
both solutions are search and edit work for me. but in old version it was not required
### Steps to reproduce

```csharp
// use the sample above ;)
```

**Comment by lbargaoanu** [comment]
Your best bet here is to fix it yourself. I suppose there are other workarounds, but I'm guessing that's not what you want.

**Comment by mjalil** [comment]
@lbargaoanu do you mean to assume this as an AutoMapper Bug and fix it? its fix is just removing one line from MappingExpression.cs
https://github.com/AutoMapper/AutoMapper/blame/49e81dff7801259be2779f3054161c3a8b6cedaf/src/AutoMapper/Internal/MappingExpression.cs#L616. OK, I can do that and send a pull request.
But if you mean to search and edit my code, that is very hard. I used `IgnoreMap` attribute everywhere. also, if I want to change mapper config like I wrote in the Actual Behavior, that is bad `.ForMember(entity => entity.Publisher, opt => opt.MapFrom(dto => dto.Publisher))` . why do I need to map publisher to publisher while AutoMapper should do that

**Comment by lbargaoanu** [comment]
You cannot remove that line :) It might break a lot of people. I just meant the second item. It shouldn't throw. That one is a bug.

**Comment by mjalil** [comment]
OK, I will fix the second item and send a pull request.
but what about the first item: silently ignoring map in reverse order. maybe someone wants to just ignore in one direction and not in the reverse. I think this is a bug too, it should not be such that.

**Comment by lbargaoanu** [comment]
Obviously someone disagreed :) That's the behavior for some time now and I don't think it makes sense to change it again.

**Comment by lbargaoanu** [comment]
If you ignore it in code, you get the behavior you want.

**Comment by lbargaoanu** [comment]
Fixed by #2273.
